### PR TITLE
Introduce helper functions vkb::[common::]select_surface_format

### LINKED ISSUE #647: Introduce helper function for selection of VKSurfaceFormatKHR
With #622, @SaschaWillems improved the selection of VkSurfaceFormatKHR in the API sample hello_triangle.
Should we move that code into some helper function and call that wherever it is appropriate?

### PR DESCRIPTION
**Pull request by asuessenbach**
## Description

It turned out, the surface format selection boilerplate was just used three times over all the samples. And the framework does it slightly differently.
But here are the helper functions to select a surface format (both for Vulkan and Vulkan-Hpp) and their usage in the samples.

Compiled and tested on Win10, using NVIDIA GPU.

Fixes #647

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
  
## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)
- [x] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](./../samples/extensions/conditional_rendering)

### CODE REVIEW

File: samples/api/hello_triangle/hello_triangle.cpp [hunk 1]
```diff
@@ -435,24 +435,7 @@ void HelloTriangle::init_swapchain(Context &context)
...
-
-	for (auto &candidate : supported_surface_formats)
-	{
-		if (std::find(preferred_format_list.begin(), preferred_format_list.end(), candidate.format) != preferred_format_list.end())
-		{
-			format = candidate;
-			break;
-		}
-	}
+	VkSurfaceFormatKHR format = vkb::select_surface_format(context.gpu, context.surface);
```
Comment (SaschaWillems) [inline samples/api/hello_triangle/hello_triangle.cpp:438]: Note sure if we should remove this part. The hello_triangle samples should is intended to be as explicit as possible, so it's probably better to put the whole code from `vkb::select_surface_format` in here.

File: samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp [hunk 2]
```diff
@@ -723,20 +723,7 @@ void HPPHelloTriangle::init_swapchain()
...
-
-	// Look for the first supported format in our list of preferred formats
-	auto formatIt =
-	    std::find_if(supported_surface_formats.begin(),
-	                 supported_surface_formats.end(),
-	                 [&preferred_format_list](vk::SurfaceFormatKHR const &candidate) { return std::find(preferred_format_list.begin(), preferred_format_list.end(), candidate.format) != preferred_format_list.end(); });
-
-	// We use the first supported format as a fallback in case none of the preferred formats is available
-	vk::SurfaceFormatKHR surface_format = (formatIt == supported_surface_formats.end()) ? supported_surface_formats[0] : formatIt->format;
+	vk::SurfaceFormatKHR surface_format = vkb::common::select_surface_format(gpu, surface);
```
Comment (SaschaWillems) [inline samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp:726]: Same as for the non-hpp hello_triangle sample.

**Review by SaschaWillems — CHANGES_REQUESTED** [review]

File: samples/api/hello_triangle/hello_triangle.cpp (hunk 1 above)
Comment (SaschaWillems) [reply samples/api/hello_triangle/hello_triangle.cpp:438]: As per today's call it's fine if the function is used here.

File: samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp (hunk 2 above)
Comment (SaschaWillems) [reply samples/api/hpp_hello_triangle/hpp_hello_triangle.cpp:726]: As per today's call it's fine if the function is used here.

**Review by tomadamatkinson — APPROVED** [review]

**Review by gpx1000 — APPROVED** [review]

**Review by SaschaWillems — APPROVED** [review]

**Pull request merged**